rust/src/lib.rs

```diff
@@ -80,21 +80,17 @@ mod static_iter {
                 );
                 return None;
             }
-            let mut static_var = STATIC_ITERATORS.lock().unwrap();
-            match &mut *static_var {
+            match &mut *STATIC_ITERATORS.lock().unwrap() {
                 None => {
-                    println!("The static_index was not found among the STATIC_ITERATORS.");
-                    None
+                    // A single thread is touching STATIC_ITERATORS.
+                    unreachable!("STATIC_ITERATORS mutex was poisoned.")
                 }
```

Comment:
That's still not it :-) The poison thing is before the `unwrap()`. Here it's just that it was not initialized, but if you use `LazyLock<Mutex<HashMap>>` then this code path disappears.